docs/apis/programming_guide.md

```diff
@@ -187,7 +187,17 @@ that creates the type information for Flink operations.
 </div>
 </div>
 
+#### Scala Dependency Versions
 
+Because Scala 2.10 binary is not compatible with Scala 2.11 binary, we provide multiple artifacts
+to support both Scala versions. If you want to run your program on Flink with Scala 2.11, you need
+to add a suffix `_2.11` to all Flink artifact ids in your dependencies. You should be careful with
+this difference of artifact id. All modules with Scala 2.11 have a suffix `_2.11` in artifact id.
+For example, `flink-java` should be changed to `flink-java_2.11` and `flink-clients` should be
+changed to `flink-clients_2.11`.
```

Comment:
Because Scala 2.10 binary is not compatible with Scala 2.11 binary, we provide multiple artifacts
to support both Scala versions. 

Starting from the 0.10 line, we cross-build all Scala-dependent Flink modules for both 2.10 and 2.11.
If you want to run your program on Flink with Scala 2.11, you need to add a `_2.11` suffix to the `artifactId` values of the Scala-dependent Flink modules in your `dependencies` section.